lib/vintage_net_mobile/modem/automatic/discovery.ex

```diff
@@ -0,0 +1,123 @@
+defmodule VintageNetMobile.Modem.Automatic.Discovery do
+  @moduledoc false
+
+  # A process for trying to discover a the type of modem and some basic
+  # information to allow dynamic configuration items
+
+  use GenServer
+  require Logger
+
+  alias Circuits.UART
+
+  # max backoff for trying to enumerate the UARTS
+  @max_backoff 3_000
```

Comment:
With a max backoff time of 3 seconds, how about deleting the backoff code and just hardcoding the poll time to be 3 seconds? I'm not sure anyone will notice and it will be one less thing that could possibly go wrong.

Reply:
Sounds good to me. Also, as a note the 3 seconds is arbitrary, so I am open to other options.